### PR TITLE
Move iOS Testing Resources to tools/internal

### PR DESCRIPTION
**Pull request by DanielToby**
This PR serves to abstract the iOS testing strategy away from any one package. Currently iOS tests performed via ci-ios.yaml are tightly coupled with core-backend. By moving shared resources to tools/internal we can begin testing several other packages, ideally with only minor additions to package.json.

The "unsolved" part of this PR is in the invocation of webpack across packages. The ios.webpack.config (now in tools/internal/ios) is configured for core-backend. The path passed into `glob.sync` is what should be variable (see line 13 of ios.webpack.config). If we can find a way to pass this value when invoking webpack, we can avoid copying ios.webpack.config for each package we'd like to test.

The ci-ios.yaml pipeline has also been improved. Because the process of running core-test-runner in App Center is largely the same between packages, that portion of the pipeline has been made into a template.

This PR also includes bug fixes to tests in core-backend. Please see [core/backend/src/test/schema/FunctionalDomain.test.ts](https://github.com/iTwin/itwinjs-core/pull/3461/files#diff-f6ad2d8198735136833a60a139d81c105e3af4a2862ec35e1b648d8403ebff14) and [core/backend/src/test/CloudStorage.test.ts](https://github.com/iTwin/itwinjs-core/pull/3461/files#diff-ac124755562a57635692b8342cf7f9015549d677106bc2af82760ba041b11410).

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: tools/internal/ios/scripts/processAppCenterLogs.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ const api_token = process.argv[4];  // app_center_token;
 const test_id = require("../run_output.json")[0].testRunId; // TODO: Pass test results path
 
 const xmlFilter = "[Mocha_Result_XML]: ";
-const deviceLogsPath = "ios/device_logs.txt"
-const resultsPath = "ios/junit_results.xml"
+const deviceLogsPath = "device_logs.txt"
```
Comment (calebmshafer) [inline tools/internal/ios/scripts/processAppCenterLogs.js:20]: where do these files end up?

Comment (DanielToby) [reply tools/internal/ios/scripts/processAppCenterLogs.js:20]: @calebmshafer in the "ios" directory. Would it be better to keep them out of the source files and instead store in lib?

Comment (calebmshafer) [reply tools/internal/ios/scripts/processAppCenterLogs.js:20]: I think storing it in lib would be fine, just wanted to make sure it went under each package's lib folder and not in some other one.

Comment (DanielToby) [reply tools/internal/ios/scripts/processAppCenterLogs.js:21]: I added a template parameter and updated the scripts to make these paths clearer.

File: tools/internal/package.json [hunk 2]
```diff
@@ -23,5 +23,8 @@
   "dependencies": {
     "yargs": "^16.0.0"
   },
-  "devDependencies": {}
+  "devDependencies": {
+    "@itwin/build-tools": "workspace:*",
```
Comment (calebmshafer) [inline tools/internal/package.json:27]: What's the reason for the `@itwin/build-tools` dep?

Comment (DanielToby) [reply tools/internal/package.json:27]: That's for access to the BentleyMochaReporter.

**Comment by calebmshafer** [comment]
```
  1) ToolRegistry
       Should parse quoted arguments with embedded quotes:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

@DanielToby Looks like there's another one... Can you increase it's timeout as well?

**Review by pmconne — APPROVED** [review]

**Pull request merged**